File: DatingAPI/Entities/Member.cs
using System.ComponentModel.DataAnnotations.Schema;
using System.Text.Json.Serialization;

namespace DatingAPI.Entities
{
    public class Member
    {
        public string Id { get; set; } = null!;
        public DateOnly DateOfBirth { get; set; }
        public string? ImageUrl { get; set; }
        public required string DisplayName { get; set; }
        public DateTime Created { get; set; } = DateTime.UtcNow;
        public DateTime LastActive { get; set; } = DateTime.UtcNow;
        public string Gender { get; set; } = null!;
        public string Description { get; set; } = null!;
        public required string City { get; set; }
        public required string Country { get; set; }

        // Navigation property
        [JsonIgnore]
        public ICollection<Photo> Photos { get; set; } = [];

        // Navigation property to AppUser   
         [JsonIgnore]
        [ForeignKey(nameof(Id))]
        public AppUser? AppUser { get; set; }
    }

}

File: client/src/types/member.ts
export type Member = {
    id: string;
     email: string;
    dateOfBirth: string; // ISO date string
    imageUrl: string;
    displayName: string;
    created: string; // ISO date string
    lastActive: string; // ISO date string
    gender: string;
    description?: string;
    city: string;
    country: string;
   
};

export type Photo = {
    id: number;
    url: string;
    memberId: string;
    publicId?: string;
    isMain: boolean;
};

export type EditableMember = {
    displayName: string;
    dateOfBirth: string; // ISO date string
    city: string;
    country: string;
    gender: string;
    description?: string;
};
